File: Python3/3083_existence_of_a_substring_in_a_string_and_its_reverse.py
# https://leetcode.cn/problems/existence-of-a-substring-in-a-string-and-its-reverse/

class Solution:
    def isSubstringPresent(self, s: str) -> bool:
        revs = set()
        for i in range(len(s)-1):
            sub = s[i:i+2]
            revs.add(sub[::-1])
            if sub in revs:
                return True
        return False

File: Cpp/3083_existence_of_a_substring_in_a_string_and_its_reverse.cpp
// https://leetcode.cn/problems/existence-of-a-substring-in-a-string-and-its-reverse/

class Solution {
public:
    bool isSubstringPresent(string s) {
        unordered_set<string> revs;
        for (int i = 0; i < s.size() - 1; ++i) {
            string sub = s.substr(i, 2);
            string sub_rev(sub);
            reverse(sub_rev.begin(), sub_rev.end());
            revs.insert(sub_rev);
            if (revs.find(sub) != revs.end()) {
                return true;
            }
        }
        return false;
    }
};